client/src/pages/Diary.jsx

```diff
@@ -0,0 +1,148 @@
+    const [message, setMessage] = useState('');
+
+    useEffect(() => {
+        const getFetch = async () => {
+            const response = await fetch(`http://localhost:3001/groups/user/${user._id}`, authHeader);
+            const data = await response.json();
+            if (response.ok){
+                setGroups(data);
+            }
+            const res_diary = await fetch(`http://localhost:3001/diary/user/${user._id}`, authHeader);
+            const data_diary = await res_diary.json();
+            if (res_diary.ok){
+                setDiaries(data_diary);
+            }
+
+            // const quill = new Quill('#editor', {
+            //     theme: 'snow'
+            // });
+        }
+        if (user){
+            getFetch();
+        }
+    }, []);
+
+    const toggleModal = () => {
+        setIsOpen(!isOpen);
+    };
+
+    const createDiary = async (event) => {
+        event.preventDefault();
+        setMessage('');
+        try {
+            const res = await axios.post("http://localhost:3001/diary/",
+                {
+                    name: newDiaryName,
+                    admin: user._id,
+                }, authHeader);
+            setMessage("New entry added successfully");
+            setNewDiaryName("Untitled");
+            toggleModal();
+        } catch (error) {
+            setMessage(error.response.data["message"]);
+        }
+    };
+
+    return (
+        <div className="h-screen">
+            <Navbar />
+            <div className="pt-[70px] h-[90%]">
+                <div class="overflow-x-auto h-[8vh] block">
+                    <button className="text-xl bg-[#FFB302] rounded-[10px] h-[6vh] p-[1vh] m-[1vh] hover:cursor-pointer hover:text-[white] hover:bg-[rgb(59,59,59)]">Personal</button>
+                    {
+                        groups.map((group) => 
+                        <button className="text-xl bg-[#FFB302] rounded-[10px] h-[6vh] p-[1vh] m-[1vh] hover:cursor-pointer hover:text-[white] hover:bg-[rgb(59,59,59)]"
+                            >
+                            {group["name"]}
+                        </button>
+                        )
+                    }
+                </div>
+                <div className="flex flex-row py-[1vh] px-[3vh] h-[100%]">
+                    <div className="w-[30%] h-[100%] mr-[1.5vh] bg-[#FFF] rounded-[20px] p-[2vh]">
+                        <div className="flex flex-row justify-center items-center">
+                            <p className="text-3xl font-bold hidden lg:block mr-[5vh]">Apr 2024</p>
+                            <button className="text-xl bg-[#FFB302] rounded-[10px] h-[6vh] py-[1vh] px-[3vh] hover:cursor-pointer hover:text-[white] hover:bg-[rgb(59,59,59)]"
+                                onClick={toggleModal}>
+                                New Entry
+                            </button>
+                            <Modal className="flex justify-center items-center"
```

Comment:
I will change this Modal component later :'D

Reply:
shadcn has a built in component for modal called Tooltip. Please take a look at my code in ChatList.jsx for reference